### PR TITLE
Eclass eapi

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

File: eclass/toolchain.eclass [hunk 1]
```diff
@@ -131,7 +130,6 @@ RESTRICT="!test? ( test )"
 
 tc_supports_dostrip() {
```
Comment (mgorny) [inline eclass/toolchain.eclass:131]: Inline it?

File: eclass/toolchain.eclass [hunk 2]
```diff
@@ -447,20 +441,12 @@ toolchain_src_unpack() {
 
 #---->> src_prepare <<----
 
-# 'epatch' is not available in EAPI=7. Abstract away patchset application
-# until we eventually get all gcc ebuilds on EAPI=7 or later.
 tc_apply_patches() {
```
Comment (mgorny) [inline eclass/toolchain.eclass:444]: Likewise.

**Review by mgorny — APPROVED** [review]
LGTM

File: eclass/toolchain.eclass [hunk 3]
```diff
@@ -447,22 +432,6 @@ toolchain_src_unpack() {
...
-		# breaking existing ebuilds use 'eapply' only in EAPI=7 or later.
-		5|6) epatch "$@" ;;
-		7|8) eapply "$@" ;;
-		*) die "Update apply_patches() for ${EAPI}." ;;
-	esac
-}
-
 toolchain_src_prepare() {
 	export BRANDING_GCC_PKGVERSION="Gentoo ${GCC_PVR}"
 	cd "${S}"
```
Comment (thesamesam) [inline eclass/toolchain.eclass:437]: || die

File: eclass/toolchain.eclass [hunk 4]
```diff
@@ -535,7 +500,8 @@ toolchain_src_prepare() {
 	sed -i 's|A-Za-z0-9|[:alnum:]|g' "${S}"/gcc/*.awk #215828
```
Comment (thesamesam) [inline eclass/toolchain.eclass:500]: missing || die but probably too risky to add for now :(

File: eclass/toolchain.eclass [hunk 5]
```diff
@@ -577,7 +545,8 @@ do_gcc_PIE_patches() {
 	want_pie || return 0
 	use vanilla && return 0
 
-	tc_apply_patches "Applying pie patches ..." "${WORKDIR}"/piepatch/*.patch
+	einfo "Applying pie patches ..."
+	eapply "${WORKDIR}"/piepatch/*.patch
 
 	BRANDING_GCC_PKGVERSION="${BRANDING_GCC_PKGVERSION}, pie-${PIE_VER}"
 }
```
Comment (thesamesam) [inline eclass/toolchain.eclass:552]: Let's modernise the readarray bit below now we're req'ing EAPI 7+.

File: eclass/toolchain.eclass [hunk 6]
```diff
@@ -1744,7 +1714,7 @@ toolchain_src_install() {
 	#  - "${D}${LIBPATH}"
 	# As dostrip does not specify host to override ${CHOST} tools just skip
 	# non-native binary stripping.
-	is_crosscompile && tc_supports_dostrip && dostrip -x "${LIBPATH}"
+	is_crosscompile && dostrip -x "${LIBPATH}"
 
 	cd "${S}"
```
Comment (thesamesam) [inline eclass/toolchain.eclass:1719]: || die

File: eclass/toolchain.eclass (hunk 5 above)
Comment (SoapGentoo) [reply eclass/toolchain.eclass:552]: bash-4.4 requires EAPI >= 8

Comment (thesamesam) [reply eclass/toolchain.eclass:552]: oops!

**Review by thesamesam — APPROVED** [review]

File: eclass/toolchain.eclass [hunk 7]
```diff
@@ -595,7 +564,8 @@ do_gcc_CYGWINPORTS_patches() {
 			echo "${d}/${p}"
 		done
 	) )
-	tc_apply_patches "Applying cygwin port patches ..." ${patches[*]}
+	einfo "Applying cygwin port patches ..."
+	eapply ${patches[*]}
```
Comment (ulm) [inline eclass/toolchain.eclass:568]: This looks suspicious. If `patches` is an array of patches, I'd expect something like this:
```suggestion
	eapply -- "${patches[@]}"
```

File: eclass/toolchain.eclass [hunk 8]
```diff
@@ -129,46 +128,36 @@ LICENSE="GPL-3+ LGPL-3+ || ( GPL-3+ libgcc libstdc++ gcc-runtime-library-excepti
...
-
 TC_FEATURES=()
 
 tc_has_feature() {
 	has "$1" "${TC_FEATURES[@]}"
 }
 
 if [[ ${PN} != "kgcc64" && ${PN} != gcc-* ]] ; then
-	IUSE+=" debug +cxx +nptl" TC_FEATURES+=(nptl)
+	IUSE+=" debug +cxx +nptl" TC_FEATURES+=( nptl )
```
Comment (ulm) [inline eclass/toolchain.eclass:138]: I'd rather not clutter the history with purely cosmetic whitespace changes (here and all following instances).

But YMMV, not a blocker for this patch.

**Review by ulm — APPROVED** [review]

**Pull request closed without merging**